### PR TITLE
Added empty state when nothing is deployed

### PR DESCRIPTION
**Pull request by milindk8**
# Description

Empty state UI is updated for Appdetails when nothing is configure & deployed

Fixes - https://github.com/devtron-labs/devtron/issues/1958

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/app/details/appDetails/AppDetails.tsx [hunk 1]
```diff
@@ -313,107 +314,139 @@ export const Details: React.FC<{
...
+                            />
+                            <ConfirmationDialog.Body
+                                title={`${hibernateConfirmationModal === 'hibernate' ? 'Hibernate' : 'Restore'} '${
+                                    appDetails.appName
+                                }' on '${appDetails.environmentName}'`}
+                                subtitle={
+                                    <p>
+                                        Pods for this application will be{' '}
+                                        <b>
+                                            scaled{' '}
```
Comment (shivani170) [inline src/components/app/details/appDetails/AppDetails.tsx:404]: Remove {' '}

File: src/components/app/details/appDetails/AppDetails.tsx [hunk 2]
```diff
@@ -850,24 +883,29 @@ export function AppNotConfigured({
...
     buttonTitle,
     appConfigTabs ='',
+    style
 }: {
     image?: any
     title?: string
     subtitle?: string
     buttonTitle?: string
     appConfigTabs?: string
+    style?:{}
```
Comment (vivek-devtron) [inline src/components/app/details/appDetails/AppDetails.tsx:893]: Please provide type

Comment (milindk8) [reply src/components/app/details/appDetails/AppDetails.tsx:893]: sure

File: src/components/app/details/appDetails/AppDetails.tsx [hunk 3]
```diff
@@ -850,24 +883,29 @@ export function AppNotConfigured({
...
 }) {
     const { appId } = useParams<{ appId: string }>()
     const { push } = useHistory()
     function handleEditApp(e) {
         getAppConfigStatus(+appId).then((response) => {
-            let url = `/app/${appId}/edit${appConfigTabs}`
+            let url = `/app/${appId}/edit`
+            if(appConfigTabs){
+                url = `/app/${appId}/${appConfigTabs}`
```
Comment (shivani170) [inline src/components/app/details/appDetails/AppDetails.tsx:901]: Earlier  url = `/app/${appId}/edit${appConfigTabs}` , As per recent changes you have removed `edit` from URL . Is this known?

File: src/components/app/details/appDetails/AppDetails.tsx [hunk 4]
```diff
@@ -313,107 +314,139 @@ export const Details: React.FC<{
...
+                                subtitle={
+                                    <p>
+                                        Pods for this application will be{' '}
+                                        <b>
+                                            scaled{' '}
+                                            {hibernateConfirmationModal === 'hibernate'
+                                                ? 'down to 0'
+                                                : ' upto its original count'}{' '}
+                                            on {appDetails.environmentName}
+                                        </b>{' '}
```
Comment (shivani170) [inline src/components/app/details/appDetails/AppDetails.tsx:409]: Same

File: src/components/app/details/appDetails/AppDetails.tsx [hunk 5]
```diff
@@ -313,107 +314,139 @@ export const Details: React.FC<{
...
+                                    appDetails.appName
+                                }' on '${appDetails.environmentName}'`}
+                                subtitle={
+                                    <p>
+                                        Pods for this application will be{' '}
+                                        <b>
+                                            scaled{' '}
+                                            {hibernateConfirmationModal === 'hibernate'
+                                                ? 'down to 0'
+                                                : ' upto its original count'}{' '}
```
Comment (shivani170) [inline src/components/app/details/appDetails/AppDetails.tsx:407]: Same

File: src/components/app/details/appDetails/AppDetails.tsx (hunk 1 above)
Comment (sohel-rp) [reply src/components/app/details/appDetails/AppDetails.tsx:404]: use `&nbsp;` for spacing

File: src/components/app/details/appDetails/AppDetails.tsx (hunk 3 above)
Comment (milindk8) [reply src/components/app/details/appDetails/AppDetails.tsx:901]: We are redirecting user to deploy & build tab. Hence removing edit.

File: src/components/app/details/appDetails/AppDetails.tsx (hunk 1 above)
Comment (milindk8) [reply src/components/app/details/appDetails/AppDetails.tsx]: removed

File: src/components/app/details/appDetails/AppDetails.tsx [hunk 6]
```diff
@@ -312,108 +313,138 @@ export const Details: React.FC<{
...
+                                            {hibernateConfirmationModal === 'hibernate'
+                                                ? 'down to 0'
+                                                : ' upto its original count'}
+                                            on {appDetails.environmentName}
+                                        </b>
+                                        environment.
+                                    </p>
+                                }
+                            >
+                                <p style={{ marginTop: '16px' }}>Are you sure you want to continue?</p>
```
Comment (sohel-rp) [inline src/components/app/details/appDetails/AppDetails.tsx:429]: use class name mt-16 instead

File: src/components/app/details/appDetails/AppDetails.tsx [hunk 7]
```diff
@@ -312,108 +313,138 @@ export const Details: React.FC<{
...
+                            message={message}
+                            nodes={aggregatedNodes}
+                            streamData={streamData}
+                            status={appDetails?.resourceTree?.status}
+                            close={(e) => toggleDetailedStatus(false)}
+                            appName={appDetails.appName}
+                            environmentName={appDetails.environmentName}
+                        />
+                    )}
+                    {showScanDetailsModal ? (
```
Comment (sohel-rp) [inline src/components/app/details/appDetails/AppDetails.tsx:386]: Also, use && here

Comment (milindk8) [reply src/components/app/details/appDetails/AppDetails.tsx]: Done

File: src/components/app/details/appDetails/AppDetails.tsx (hunk 6 above)
Comment (milindk8) [reply src/components/app/details/appDetails/AppDetails.tsx]: Changed

File: src/components/app/details/appDetails/AppDetails.tsx [hunk 8]
```diff
@@ -890,7 +889,7 @@ export function AppNotConfigured({
     subtitle?: string
     buttonTitle?: string
     appConfigTabs?: string
-    style?:{}
+    style?:object
```
[the remark this answers is not in the document]
Comment (milindk8) [reply src/components/app/details/appDetails/AppDetails.tsx]: Done

**Review by sohel-rp — APPROVED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**